### PR TITLE
[AWS] Support owning account for cross account monitoring

### LINKED ISSUE #40570: [AWS] Enable OwningAccount for cross account monitoring
In `ListMetrics` API, we added `IncludeLinkedAccounts` configuration for AWS CloudWatch to enable collecting all metrics from source accounts. But we are still missing `OwningAaccount` configuration to only collect metrics from one source account. 

doc: https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_ListMetrics.html

### PR DESCRIPTION
**Pull request by Kavindu-Dodan**
## Proposed commit message

Adds `owning_account` configuration option for AWS where it accepts a source account ID. Internally, this maps to the ListMetrics API's [^1] `OwningAccount` parameter. Note that `include_linked_accounts` should be set to `true` to use the account filtering. 

```yaml
include_linked_accounts: true # default to true, hence can be ignored in configuration yaml
owning_account: 111111111111
```

See the screenshot below from an actual run with owning account filtering, 

Screenshot 2024-09-09 at 10 26 11 AM

In comparison, consider the screenshot below where both monitoring account and linked accounts are visible as there is no filtering for `owning_account`, 

Screenshot 2024-09-10 at 9 48 42 AM

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Disruptive User Impact

None :`owning_account` is a new configuration 

## How to test this PR locally

- Build x-pack metricsbeat 
- Enable AWS monitoring with `include_linked_accounts` set to true (default is already true) & `owning_account` account ID
- Observe metrics data and validate that `cloud.account.id` match with `owning_account` ID

## Related issues

Closes #40570

[^1]:  https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_ListMetrics.html#API_ListMetrics_RequestParameters

### CODE REVIEW

File: x-pack/metricbeat/modules.d/aws.yml.disabled [hunk 1]
```diff
@@ -15,6 +15,8 @@
   period: 5m
   metricsets:
     - cloudwatch
+  include_linked_accounts: true
+  owning_account: 111111111111
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/modules.d/aws.yml.disabled:19]:
```suggestion
  # owning_account: 111111111111
```
I would comment out this line just to make aws.yml easy to use (can be used directly without changing any config).

File: CHANGELOG.next.asciidoc [hunk 2]
```diff
@@ -195,6 +195,7 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - Fix Azure Monitor metric timespan to restore Storage Account PT1H metrics {issue}40376[40376] {pull}40367[40367]
 - Remove excessive info-level logs in cgroups setup {pull}40491[40491]
 - Add missing ECS Cloud fields in GCP `metrics` metricset when using `exclude_labels: true` {issue}40437[40437] {pull}40467[40467]
+- Add AWS OwningAccount support for cross account monitoring {issue}0570[0570] {pull}40691[40691]
```
[the remark this answers is not in the document]
Comment (Kavindu-Dodan) [reply CHANGELOG.next.asciidoc:198]: good catch :) thank you

File: x-pack/metricbeat/modules.d/aws.yml.disabled (hunk 1 above)
Comment (Kavindu-Dodan) [reply x-pack/metricbeat/modules.d/aws.yml.disabled:19]: Yup, that's a good point :)

**Comment by kaiyan-sheng** [comment]
@elastic/elastic-agent-data-plane Could we get a review for this PR please? Thanks!!

**Review by pierrehilbert — APPROVED** [review]
Approving for the Elastic Agent Data Plane part as this is only adding new configuration entries in the yml file.

**Comment by pierrehilbert** [comment]
I just remembered that you mentioned the target is 8.15.2: you should add the `backport-8.15` label if this is the case.

**Pull request merged**